tests/nosetests/pyanaconda_tests/module_payload_shared.py

```diff
@@ -100,51 +100,58 @@ def prepare_source(source_type, state=SourceState.READY):
         self._test.assertFalse(self.payload_interface.HasSource())
 
-    def check_set_sources(self, test_sources, exception=None, expected_sources=None):
+    def set_and_check_sources(self, test_sources, exception=None):
         """Default check to set sources.
 
         :param test_sources: list of sources for emptiness failed check
         :type test_sources: list of source instances
         :param exception: exception class which will be raised for the given sources
-        :param expected_sources: list of expected sources after trying to set;
-                                 including when exception raised
-        :type expected_sources: list of source instances
+        :return: caught exception if exception raised
+        """
+        ret = self.set_sources(test_sources, exception)
+
+        if exception:
+            self.check_sources([])
+        else:
+            self.check_sources(test_sources)
+
+        return ret
+
+    def set_sources(self, test_sources, exception=None):
+        """Set sources to payload.
+
+        :param test_sources: list of sources for emptiness failed check
+        :type test_sources: list of source instances
+        :param exception: exception class which will be raised for the given sources
         :return: caught exception if exception raised
         """
         paths = PayloadSourceContainer.to_object_path_list(test_sources)
-        ret = None
 
         if exception:
             with self._test.assertRaises(exception) as cm:
                 self.payload_interface.SetSources(paths)
-            ret = cm
-        else:
-            self.payload_interface.SetSources(paths)
+            return cm
 
-        if expected_sources:
-            expected_paths = PayloadSourceContainer.to_object_path_list(expected_sources)
+        self.payload_interface.SetSources(paths)
+        return None
+
+    def check_sources(self, expected_sources=None):
+        """Check what sources are set in payload.
 
-            self._test.assertEqual(self.payload_interface.Sources, expected_paths)
+        :param expected_sources: list of expected sources after trying to set;
+                                 including when exception raised
+        :type expected_sources: list of source instances
+        """
+        expected_paths = PayloadSourceContainer.to_object_path_list(expected_sources)
+
+        self._test.assertEqual(self.payload_interface.Sources, expected_paths)
+
+        if expected_sources:
             self._test.assertTrue(self.payload_interface.HasSource())
-        elif exception:
-            self._test.assertEqual(self.payload_interface.Sources, [])
-            self._test.assertFalse(self.payload_interface.HasSource())
         else:
-            self._test.assertEqual(self.payload_interface.Sources, paths)
-            self._test.assertTrue(self.payload_interface.HasSource())
-
-        return ret
+            self._test.assertFalse(self.payload_interface.HasSource())
```

Comment:
The DBus method `HasSource` will be removed.

Reply:
Lets do it with a different PR which will remove the `HasSource()` API.
Also why it should go away?

Reply:
It is redundant. You can get the same information by checking the `Sources` property.

Reply:
OK, I'll remove it to safe lines later.

Reply:
There is an issue for that in Jira now.